arc-s/arcs.cpp

```diff
@@ -5,3 +5,261 @@
 // $Copyright: Copyright (C) LDO Systems 2018
 //###########################################################################
 #include "arcs.h"
+#include <math.h>
+#define IOCLOCK 16000000
+
+Arcs::Arcs()
+{
+	accelerategap = 1000;
+	acceleratedspeed = 10;
+}
+
+
+///Update pulse speed
+void Arcs::updateStep()
+{
+	OCR5A = uint16_t((IOCLOCK / 2) / (lines * rev * microstep / 60) + 0.5) - 1;
+	delayMicroseconds(accelerategap);
+}
+
+
+///Initialize the pin
+///Note that the Step pin can only select one of the 46 pin(PL3), the remaining pins can be more demanding choice
+void Arcs::Initialize(ConfigPinStruct pinInfo)
+{
+	pinEn.Initialize(pinInfo.pinEnCh, Gpio::_OUTPUT, pinInfo.pinEnNum);
+	pinReset.Initialize(pinInfo.pinResetCh, Gpio::_OUTPUT, pinInfo.pinResetNum);
+	pinCurrentMode.Initialize(pinInfo.pinCurrentModeCh, Gpio::_OUTPUT, pinInfo.pinCurrentModeNum);
+	pinDir.Initialize(pinInfo.pinDirCh, Gpio::_OUTPUT, pinInfo.pinDirNum);
+	pinMs1.Initialize(pinInfo.pinMs1Ch, Gpio::_OUTPUT, pinInfo.pinMs1Num);
+	pinMs2.Initialize(pinInfo.pinMs2Ch, Gpio::_OUTPUT, pinInfo.pinMs2Num);
+	pinMs3.Initialize(pinInfo.pinMs3Ch, Gpio::_OUTPUT, pinInfo.pinMs3Num);
+	pinStep.Initialize(Gpio::_PL, Gpio::_OUTPUT, 3);
+
+	//Initialize timer 5
+	TCCR5A = 0;
+	TCCR5B = 0;
+	TCCR5A |= _BV(COM5A0) | _BV(WGM51) | _BV(WGM50);
+	TCCR5B |= _BV(WGM52) | _BV(WGM53);
+	TCCR5B |= _BV(CS50);
+}
+
+
+///Set the speed of the stepper motor
+void Arcs::configSpeed(uint32_t lines, double rev, ArcsMicroStep microstep)
```

Comment:
separate into configMotor(uint32_t stepsPerRev, ArcsMicroStep microstep) and setSpeed(double speedRPM)